src/debugger/adapter/AdapterChannel.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+import fs from "fs";
+
+const TWO_DASH = "--";
+
+export class AdapterChannel {
+  constructor(inFilePath: string, outFilePath: string) {
+    this.inFilePath = inFilePath;
+    this.outFilePath = outFilePath;
+  }
+  inFilePath: string;
+  outFilePath: string;
+
+  writeOut(contents: string) {
+    fs.writeFileSync(this.outFilePath, contents.length + TWO_DASH + contents);
+  }
+
+  listenOnFile(errorHandler: (err: ?ErrnoError) => void, messageProcessor: (message?: string) => void) {
+    fs.readFile(this.inFilePath, { encoding: "utf8" }, (err: ?ErrnoError, content: string) => {
+      if (err) {
+        errorHandler(err);
+        return;
+      }
+      // format: <length>--<contents>
+      let separatorIndex = content.indexOf(TWO_DASH);
```

Comment:
There are many duplicate/shared logic between here and DebuggerChannel read. You may consider how to share the code(like introduce another shared class which handles the message format details) so that if we would like to change the message format, one change is needed(instead two places).